src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -48,147 +55,180 @@ internal partial class DesignerAttributeIncrementalAnalyzer : ForegroundThreadAf
-            _state = new DesignerAttributeState();
-        }
-
-        public Task DocumentResetAsync(Document document, CancellationToken cancellationToken)
-        {
-            _state.Remove(document.Id);
-            return _state.PersistAsync(document, new Data(VersionStamp.Default, VersionStamp.Default, designerAttributeArgument: null), cancellationToken);
         }
 
         public bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs e)
         {
             return false;
         }
 
-        public async Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, InvocationReasons reasons, CancellationToken cancellationToken)
+        public async Task AnalyzeProjectAsync(Project project, bool semanticsChanged, InvocationReasons reasons, CancellationToken cancellationToken)
         {
-            Contract.ThrowIfFalse(document.IsFromPrimaryBranch());
-
+            Contract.ThrowIfFalse(project.IsFromPrimaryBranch());
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (!document.Project.Solution.Workspace.Options.GetOption(InternalFeatureOnOffOptions.DesignerAttributes))
+            var vsWorkspace = project.Solution.Workspace as VisualStudioWorkspaceImpl;
+            if (vsWorkspace == null)
             {
                 return;
             }
 
-            // use tree version so that things like compiler option changes are considered
-            var textVersion = await document.GetTextVersionAsync(cancellationToken).ConfigureAwait(false);
-            var projectVersion = await document.Project.GetDependentVersionAsync(cancellationToken).ConfigureAwait(false);
-            var semanticVersion = await document.Project.GetDependentSemanticVersionAsync(cancellationToken).ConfigureAwait(false);
-
-            var existingData = await _state.TryGetExistingDataAsync(document, cancellationToken).ConfigureAwait(false);
-            if (existingData != null)
+            if (!vsWorkspace.Options.GetOption(InternalFeatureOnOffOptions.DesignerAttributes))
             {
-                // check whether we can use the data as it is (can happen when re-using persisted data from previous VS session)
-                if (CheckVersions(document, textVersion, projectVersion, semanticVersion, existingData))
-                {
-                    RegisterDesignerAttribute(document, existingData.DesignerAttributeArgument);
-                    return;
-                }
+                return;
             }
 
-            var result = await ScanDesignerAttributesOnRemoteHostIfPossibleAsync(document, cancellationToken).ConfigureAwait(false);
-            if (result.NotApplicable)
+            // CPS projects do not support designer attributes.  So we just skip these projects entirely.
+            var isCPSProject = await Task.Factory.StartNew(
+                () => vsWorkspace.IsCPSProject(project),
```

Comment:
Note: i've currently removed this check until we can decide what to do here.  I'd like to get the rest of the improvements in before that.